design-proposals/best-match-cpu-model.md

```diff
@@ -0,0 +1,165 @@
+as nodes are being added and removed from the cluster.
+
+It isn't a trivial task for a user to calculate optimal CPU Model that will be relatively new and
+won't prevent VMIs from being live migrated.
+
+Even if a user has this information he would have to set the nodeAffinity himself and as mentioned
+this might cause him to ignore some nodes that don't support this optimal CPU Model. 
+
+Therefore, some users might benefit from such configuration options.
+
+## Goals
+- Dynamically choose a CPU Model that is relatively new and will allow users with heterogeneous 
+cluster to live migrate VMIs.
+
+- Avoid Ignoring some of the nodes in a cluster as it can cause issues with resource 
+allocation and availability. it could lead to over-utilization of other nodes and potential performance issues. 
+
+
+## Non Goals
+Not changing the default form in which we schedule virtual machines.
+
+## Definition of Users
+Every user that is interested in migration of VMIs in heterogeneous cluster.
+
+## User Stories
+- As a user with a long living heterogeneous cluster, I'm **SURPRISED** to encounter `host-model` migration 
+are always pending and i don't know what can i do to migrate these VMIs.
+
+- As a user that use KubeVirt in heterogeneous cluster i set the CPU model to be
+`Opteron_G1` because it is supported in all the cluster nodes and live migration is 
+important to my use cases, but I realized that `Opteron_G1` CPU model does not include 
+some of the features and capabilities of newer CPU models, such as support for virtualization 
+extensions like Intel VT-x or AMD-V, which are required for running certain types 
+of my workloads.
+
+- As a user that use KubeVirt in heterogeneous cluster i set the CPU model to be
+`Skylake-Client-IBRS` because it relatively new and performance is important to 
+my workloads, but I realized that my VMIs always land on the same set 
+of nodes  because some of the nodes don't support `Skylake-Client-IBRS`.
+
+Configuration option that calculate best match for CPU Model dynamically after a vmi is scheduled 
+to a node can help these users.
+
+## Repos
+https://github.com/kubevirt/kubevirt
+
+# Design
+- We will hold a `map[string]string` from CPU Model to launch year ,this heuristic will allow
+us choosing relatively new CPU Model when using CPU Model matcher.
+
+- When using the CPU Model matcher a VMI's CPU Model will be calculated **AFTER** virt-launcher is
+scheduled to a node.
+
+**Calculation** : First we will get all the potential nodes that respect the VMI's node affinity, 
+node selectors and has the same vendor as the initial node.  
+Then for each CPU Model that has newer launch year then `MinimalLaunchYear` that is supported 
+on the initial node we will calculate how many potential nodes support it.  
+We will define a `ThresholdSupport` and we will choose the CPU Model that is
+supported by at least `ThresholdSupport` of the cluster's node with 
+the newest launch year.
```

Comment:
What if multiple CPU models have the same rank, i.e. supported by the same amount of nodes? Would we take the oldest model?